### PR TITLE
DHFPROD-7763: Table-view

### PR DESCRIPTION
**Pull request by iveyMarklogic**
#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Code passes ESLint tests
- [x] Added Tests
- [x] Run UI tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

File: marklogic-data-hub-central/ui/src/components/common/hc-table/hc-table.tsx [hunk 1]
```diff
@@ -50,12 +50,12 @@ function HCTable({className, rowStyle, childrenIndent, data, expandedRowKeys, ne
...
         {expanded ?
-          <ChevronDown className={styles.iconIndicator} aria-label="down"/> :
-          <ChevronRight className={styles.iconIndicator} aria-label="right"/>}
+          <ChevronDown className={styles.iconIndicator} aria-label="down" /> :
+          <ChevronRight className={styles.iconIndicator} aria-label="right" />}
       </HCButton>;
     },
     expandHeaderColumnRenderer: () => "",
-    nonExpandable: !props.subTableHeader && !childrenIndent ? [] : data.map(row => typeof rowKey === "string" && row[rowKey] && !row.children && row[rowKey]).filter(row => !!row),
+    nonExpandable: !props.subTableHeader && !childrenIndent ? [] : data.filter(row => typeof rowKey === "string" && !row.children).map(element => typeof rowKey === "string"  && typeof element[rowKey] !== undefined && element[rowKey]),
```
Comment (mrodrigu-marklogic) [inline marklogic-data-hub-central/ui/src/components/common/hc-table/hc-table.tsx:58]: This is the bit I told you earlier @lucianaabc-ml

**Review by mrodrigu-marklogic — APPROVED** [review]

**Review by MSBrignoni — CHANGES_REQUESTED** [review]
Nice work @iveyMarklogic, just a comments, perhaps some of these will be fixed in the near future globally:
- the header color is different in this table in nightly.
- there are some hovers effects and different grey in the headers of the other tables (collections table and the others).

color header table

image

File: marklogic-data-hub-central/ui/src/pages/Bootstrap.scss [hunk 2]
```diff
@@ -1,33 +1,33 @@
 /* TODO Test view for React Bootstrap components, Bootstrap.scss TO BE REMOVED */
 
-.mosaic-container-bootstrap .mosaic-window-body {
-    padding: 20px;
-}
+// .mosaic-container-bootstrap .mosaic-window-body {
+//     padding: 20px;
+// }
```
Comment (yubiselmarklogic) [inline marklogic-data-hub-central/ui/src/pages/Bootstrap.scss:5]: @iveyMarklogic  Consider not leaving unnecessary comments in the code

**Review by yubiselmarklogic — DISMISSED** [review]

File: marklogic-data-hub-central/ui/src/pages/Bootstrap.scss (hunk 2 above)
Comment (iveyMarklogic) [reply marklogic-data-hub-central/ui/src/pages/Bootstrap.scss]: @yubiselmarklogic i removed this coments

**Comment by iveyMarklogic** [comment]
@MSBrignoni I fixed both comments

**Review by MSBrignoni — APPROVED** [review]

**Review by yubiselmarklogic — DISMISSED** [review]

**Review by briantang — DISMISSED** [review]
Looks good now! Thanks, @iveyMarklogic

**Review by yubiselmarklogic — APPROVED** [review]

**Review by MSBrignoni — APPROVED** [review]

**Pull request merged**